index.bs

```diff
@@ -3763,10 +3772,653 @@ nothrow>WritableStreamDefaultControllerError ( <var>controller</var>, <var>error
+      written to the writable side to result in zero or multiple chunks on the readable side.
+
+      <p>The stream implementation guarantees that <code>transform</code> will be called only after previous
+      transformations have succeeded, and never before <code>start</code> has completed or after <code>flush</code> is
+      called. When no <code>transform</code> method is supplied, the identity transform is used, which enqueues chunks
+      unchanged from the writable side to the readable side.
+
+    <li><p><code>flush(controller)</code> is called after all chunks written to the writable side have been
+      transformed and the writable side has been closed. It is typically used to enqueue suffix chunks to the
+      readable side, before that too becomes closed. If this process is asynchronous, it can return a promise
+      to signal success or failure.
+  </ul>
+
+  The <code>controller</code> object passed to <code>start</code>, <code>transform</code> and <code>flush</code> is an
+  instance of {{TransformStreamDefaultController}}, and has the ability to enqueue chunks to the readable side,
+  or to terminate or error the stream.
+
+  The second and third arguments to the constructor are the <a>queuing strategy</a> objects for the writable and
+  readable sides respectively. These are used in the construction of the {{WritableStream}} and {{ReadableStream}}
+  objects and can be used to add buffering to a {{TransformStream}}, in order to smooth out variations in the speed of
+  the transformation, or to increase the amount of buffering in a <a>pipe</a>.
+</div>
+
+<emu-alg>
+  1. Set *this*.[[transformer]] to _transformer_.
+  1. Set *this*.[[transformStreamController]] to *undefined*.
+  1. Set *this*.[[backpressure]] and *this*.[[backpressureChangePromise]] to *undefined*.
+  1. Let _readableType_ be ? GetV(_transformer_, `"readableType"`).
+  1. If _readableType_ is not *undefined*, throw a *RangeError* exception.
+  1. Let _writableType_ be ? GetV(_transformer_, `"writableType"`).
+  1. If _writableType_ is not *undefined*, throw a *RangeError* exception.
+  1. Let _controller_ be  ? Construct(`<a idl>TransformStreamDefaultController</a>`, « *this* »).
+  1. Set *this*.[[transformStreamController]] to _controller_.
+  1. Let _startPromise_ be <a>a new promise</a>.
+  1. Let _source_ be ! Construct(`<a idl>TransformStreamDefaultSource</a>`, « *this*, _startPromise_ »).
+  1. Let _readableStrategy_ be ! ObjectCreate(<a>%ObjectProperty%</a>).
+  1. Perform ! CreateDataProperty(_readableStrategy_, `"size"`, _size_).
+  1. Perform ! CreateDataProperty(_readableStrategy_, `"highWaterMark"`, _highWaterMark_).
+  1. Set *this*.[[readable]] to ? Construct(`<a idl>ReadableStream</a>`, « _source_, _readableStrategy_ »).
+  1. Let _sink_ be ! Construct(`<a idl>TransformStreamDefaultSink</a>`, « *this*, _startPromise_ »).
+  1. Set *this*.[[writable]] to ? Construct(`<a idl>WritableStream</a>`, « _sink_, _writableStrategy_ »).
```

Comment:
missing _writableStrategy_ creation

Reply:
It comes directly from the constructor arguments. I can add a note to explain the asymmetry between readableStrategy and writableStrategy.

Reply:
Oops, sorry. I should've checked the arguments list.

Yeah, it might be helpful, thought it's just because I'm not catching up with the changes. Is https://github.com/whatwg/streams/commit/fd17cf00133784f5a8ed251e8a0fe633967875a8 the only background of the asymmetry?

Reply:
Yes, and issue #777 which discusses it.